src/ast/ast_build_filter_tree.c

```
@@ -28,348 +29,53 @@ void _FT_Append(FT_FilterNode **root_ptr, FT_FilterNode *child) {
-FT_FilterNode *AST_BuildFilterTreeFromClauses
-(
-	const AST *ast,
-	const cypher_astnode_t **clauses,
-	uint count
-) {
-	cypher_astnode_type_t type;
-	FT_FilterNode *filter_tree = NULL;
-	const cypher_astnode_t *predicate = NULL;
+		// Build equality predicate: entity.prop_name = val
+		AR_ExpNode *lhs = AR_EXP_NewAttributeAccessNode(
+			AR_EXP_NewVariableOperandNode(cypher_ast_identifier_get_name(entity)),
```

Cppcheck (2.21.0)

[error] Memory leak

(memleak)